meetings/cranelift/2021/cranelift-10-04.md

```diff
@@ -14,8 +14,155 @@
+  be driven by doing the "new thing" first and falling back to the "old thing"
+  if `None` is returned. Will put ISLE in place, start with "add" or something,
+  ask for help with PRs. Individual instructions should be simple and when it's
+  all done we can remove the old way.
+* akirilov - second escape hatch!
+* cfallin - yes still have old way where we can use if ISLE can't handle
+  something and we can figure out later if we need to add something to ISLE
+* bbouvier - what if ISLE generates slower Rust than what we do today, we won't
+  figure this out with incremental rewrite and would need to compare two points
+  in time which could have other changes in the middle. Hard to measure?
+* cfallin - should have no performance regressions by construction. Should
+  generate the same code that we're writing by hand. Shouldn't have regressions
+  at this time. If many more complicated patterns arise then we should be in
+  theory generating better code. In the future we write more complicated code
+  with more complicated patterns that's a new state. Today though we should
+  strive to make sure the generated code matches what we do today.
+* akirilov - can check for regressions with current backends by not deleting
+  code in current backends and skipping ISLE to do comparison.
+* cfallin - true, can add all the things and delete separately, but the downside
+  is that we have two sources of truth with possible divergences. Have the
+  flexibility of doing that though.
+* bbouvier - risk that there is an exponential growth of pattern matching trees.
+  Could two rules interact in a way that they do deeper matching? Hard to test
+  we didn't do that in particular with incremental rewrite.
+* cfallin - could be interesting interactions, yeah, currently a property is
+  that the output of ISLE is linear in the input size. No iteration to a fixed
+  point in combining rules, it instead does only a single layer. Shouldn't get
+  exponential blowup in that regard but could be interesting interactions
+  perhaps though, but none over what we already do.
+* bbouvier - concern about verbosity - if a specific rule is commutative do you
+  need to write a rule twice? Or systemic checks about commutativity?
+* cfallin - nothing like that right now, would write it twice. Idea is to have
+  it be simple first. If we express and can lower it simply then we can do that
+  as well yeah.
+* fitzgen - I think this is a non-issue since we already have `iadd_imm`
+  canonicalized by the preopt pass, no need to re-canonicalize in lowerings.
+* cfallin - still good question if we can express a family of patterns. We could
+  have a macro system perhaps in the futures where one rule goes to N rules, but
+  maybe too much complexity too. For now cases should be simple enough.
+* fitzgen - gcc/go have macro in their DSL to be polymorphic over bit-width.
+* cfallin - for that specific case we could
+
+... alexcrichton had video issues and disconnected ...
```

Comment:
I think this was just a note about how we could be polymorphic on types by binding the type with a pattern variable.